source/common/unified_memory_allocation/CMakeLists.txt

```diff
@@ -1,11 +1,19 @@
 # Copyright (C) 2023 Intel Corporation
 # SPDX-License-Identifier: MIT
 
-add_library(unified_memory_allocation STATIC
+set(UMA_SOURCES
     src/memory_pool.c
     src/memory_provider.c
 )
 
+if(UMA_BUILD_SHARED_LIBRARY)
```

Comment:
Is `USM_BUILD_SHARED_LIBRARY` only defined when the user sets it during CMake configuration?

Making it an [option](https://cmake.org/cmake/help/latest/command/option.html) would be nice.

Reply:
Right, done.